src/normalizeWhiteSpaces.spec.ts

```diff
@@ -12,4 +12,11 @@ describe('normalizeWhiteSpaces', () => {
       'Hi, how is everything ?'
     );
   });
+
+  it('Normalize white-space in string template', () => {
+    const student = { name: 'John   Doe' };
+    expect(
+      normalizeWhiteSpaces(` Hi,  ${student.name}!\n How are you doing? `)
```

Comment:
@phuvinhbmt I also used the wrong term in the issue, its [tagged templates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates), and they're little bit different than just supporting string templates. Tagged templates are functions that receives a list os strings as first argument, and the interpolations as the rest of arguments.

```ts
function normalizeWhiteSpaces(parts: TemplateStringArray, ...values: unknown[]): string;
```

```suggestion
  it('supports usage as tagged template', () => {
    const student = { name: 'John   Doe' };
    expect(
      normalizeWhiteSpaces` Hi,  ${student.name}!\n How are you doing? `
```